fault/tester/ready_valid_tester.py

```diff
@@ -0,0 +1,52 @@
+import magma as m
+from fault.tester.staged_tester import Tester
+from fault.assert_immediate import assert_immediate
+
+
+class ReadyValidTester:
```

Comment:
It's a bit odd to call this class `ReadyValidTester` and not inherit from `BaseTester`. Ostensibly the "right" thing to do would be to inherit from `BaseTester` and then raise exceptions on all the peek/poke/etc type methods? But I understand that's a bit non-robust because everytime a method is added there, we would have to nullify it there.

Is there any reason to structure it as a class? I think it makes more sense to do the following:
* Define a separate function `wrap_with_sequence` (prob a better name than this...) that just returns a new synthesizable circuit which is the Wrapper you defined here.
* Then have one function like
```python
def run_ready_valid_test(ckt: m.DefineCircuitKind, sequences: Mapping, synthesizable: bool = True):
    if synthesizable:
        wrapped = wrap_with_sequence(ckt, sequences)
        tester = Tester(wrapped)
        tester.wait_until_high(wrapped.done)
        tester.compile_and_run(*args, **kwargs)
        return
    # some logic to run in non-synthesizable mode, maybe a For-Loop tester?
```